tests/test-extensions/classloading-extension/src/main/java/org/mule/test/classloading/ClassLoadingExtension.java

```diff
@@ -11,12 +11,14 @@
 import org.mule.runtime.extension.api.annotation.Extension;
 import org.mule.runtime.extension.api.annotation.connectivity.ConnectionProviders;
 import org.mule.runtime.extension.api.annotation.dsl.xml.Xml;
+import org.mule.runtime.extension.api.annotation.metadata.MetadataScope;
 import org.mule.test.classloading.api.ClassLoadingHelper;
 
 @Extension(name = "ClassLoading")
 @Xml(prefix = "classloading")
 @ConnectionProviders({CLPoolingConnectionProvider.class, CLCachedConnectionProvider.class, CLNoneConnectionProvider.class})
 @Configurations(CLConfiguration.class)
+@MetadataScope(keysResolver = CLKeysResolver.class)
```

Comment:
I would set the type resolver for a specific operation you want to modify, or create a new one.

Reply:
This is needed for configs to be annotated with the KeysResolver annotation in the model. That then allows the following line to be called in the tests:
`metadataService.getMetadataKeys(builder().globalName("none").build());`
and that will trigger getting the metadata keys.